src/storage/write_ahead_log/log_writer.cpp

```diff
@@ -0,0 +1,59 @@
+#include "storage/write_ahead_log/log_writer.h"
+#include "storage/write_ahead_log/log_manager.h"
+
+namespace terrier::storage {
+
+void LogWriter::FlushAllBuffers() {
+  // Persist all the filled buffers to the disk
+  while (!log_manager_->filled_buffer_queue_.Empty()) {
+    // Dequeue filled buffers and flush them to disk
+    BufferedLogWriter *buf;
+    log_manager_->filled_buffer_queue_.Dequeue(&buf);
+    buf->FlushBuffer();
+    // Enqueue the flushed buffer to the empty buffer queue
+    log_manager_->empty_buffer_queue_.Enqueue(buf);
+  }
+  // Persist the buffers
+  TERRIER_ASSERT(!(log_manager_->buffers_.empty()), "Buffers vector should not be empty until Shutdown");
+  // Force the buffers to be written to disk
+  log_manager_->buffers_.front().Persist();
+}
+
+void LogWriter::WriteToDisk() {
+  // Log writer thread spins in this loop
+  // It dequeues a filled buffer and flushes it to disk
+  while (log_manager_->run_log_writer_thread_) {
```

Comment:
As before, strip the wake-up logic and place elsewhere.

Reply:
We h

Reply:
I meant you should probably decouple the write and wakeup part of this function.

Reply:
There doesn't seem to be a straightforward way to do this. We designed this keeping in mind the writer thread upon waking up knows what action to take. If you could give a guideline, it would be really hellpful